resolver/GlobalPass.cc

```diff
@@ -345,20 +345,179 @@ void fullLinearizationSlowImpl(core::GlobalState &gs, const ParentLinearizationI
         }
     }
 };
+
```

Comment:
I only just skimmed this code but couldn't find anything:

Does it make sense to use `requires_ancestor` inside a class? I think it only makes sense in in a module.

- If it makes sense in a class, is there a test?
- If it doesn't, can we emit an error for using it in a class, and write a test?

Reply:
I added a test showing how to use it on a class: https://github.com/sorbet/sorbet/pull/3409/files#diff-deae14f8b7f473687b45d25ebbcbb820R1-R34

There are also two tests about errors with classes: https://github.com/sorbet/sorbet/pull/3409/files#diff-8619494c7637bf3655aac0a3eedfe6f0R33-R41